src/d3a/models/strategy/utils.py

```diff
@@ -0,0 +1,40 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License as published by the Free Software Foundation, either version 3 of the
+License, or(at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program. If not,
+see <http://www.gnu.org/licenses/>.
+"""
+import numpy as np
+
+from d3a.constants import DEFAULT_PRECISION, RELATIVE_STD_FROM_FORECAST_ENERGY
+
+
+def compute_altered_energy(
+        energy_kWh: float, relative_std: float = RELATIVE_STD_FROM_FORECAST_ENERGY,
+        random_generator: np.random.Generator = np.random.default_rng(seed=12)) -> float:
```

Comment:
I have one more doubt here though: Every time this function is called we initiate a new RNG. Doesn't this increase the computational effort? for the whole simulation?

Reply:
On one side, initializing this kind of instance should be very light so I don't think it would represent an issue.
But most importantly:
> Python’s default arguments are evaluated once when the function is defined, not each time the function is called.
(https://docs.python-guide.org/writing/gotchas/#what-actually-happens)

You can see it with this example:
```python
class MyClass:
   pass

def my_function(foo=MyClass()):
   """Print the ID of the actual object in memory"""
   print(id(foo))

>>> my_function()
4491117472

>>> my_function()
4491117472

>>> my_function()
4491117472
```